redaxo/src/core/lib/util/string.php

```diff
@@ -254,4 +254,17 @@ public static function sanitizeHtml(string $html): string
         /** @psalm-taint-escape html */
         return $antiXss->xss_clean($html);
     }
+
+    /**
+     * Converts a string content to a array.
+     *
+     * @param string $value
+     *
+     * @return array|null
+     */
+    public static function toArray($value)
+    {
+        $value = json_decode(htmlspecialchars_decode($value, ENT_QUOTES), true);
```

Comment:
Bin mir nicht sicher, ob ich die Methode hier so allgemein gut finde.
Eigentlich ist sie schon spezifisch für die rex-vars, bzw. slice-values. 
Gerade das `htmlspecialchars_decode` hatte spezielle Gründe, aber für eine ganz allgemeine Methode ist es eher nicht sinnvoll, vermute ich.

Auch finde ich `rex_string::toArray` so unspezifisch. Also könnte ja diverse String-Varianten von Arrays geben, hier ist es aber ja ganz speziell JSON.
Zum Beispiel haben wir schon `rex_string::split`, was auch einen String in ein Array umformt.

(`rex_var::toArray` hingegen ist weniger allgemein, denn da ist es ja spezifisch für die Array-Form gedacht, die rex_vars nun mal nutzen.)

Für welchen Anwendungsfall brauchst du die Methode?

Reply:
> Für welchen Anwendungsfall brauchst du die Methode?

In einem YForm Feld nutze ich [Tagify](https://github.com/yairEO/tagify) um Schlagworte zu speichern. Zum Schluss landet Json in der DB. 

Auch hier bei einer _aktuellen_ Migration sind in der Quelldatenbank Json Daten hinterlegt.

Fand es daher ganz nützlich, die bisherige Methode allgemein gültiger zu machen.

_Project: 542 _

Reply:
für rex_sql haben wir `$sql->setArrayValue` und `$sql->getArrayValue`.

wenn wir es hier in `rex_string` aufnehmen, würde ich "json" mit in den Methodennamen nehmen, zur Unterscheidung. 
`rex_string::jsonDecode` (haben ja auch schon `rex_string::yamlDecode`).
Eventuell auch `rex_string::jsonArrayDecode`, falls wir bewusst nur Arrays zulassen wollen (siehe auch https://github.com/redaxo/redaxo/pull/5508).

Das `htmlspecialchars_decode` sollte hier dann aber raus (entsprechend muss `rex_var` seine eigene Implementierung behalten).